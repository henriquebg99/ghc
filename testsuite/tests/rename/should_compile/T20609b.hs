module T20609b where

-- Triggers the warning (definition/binding sites):
-- ------------------------------------------------

class MyClass c where
  forall :: c -> ()

-- Does not trigger the warning (use sites):
-- -----------------------------------------

instance MyClass () where
  forall = id
