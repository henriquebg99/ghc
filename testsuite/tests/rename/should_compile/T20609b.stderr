
T20609b.hs:7:3: warning: [-Wforall-identifier (in -Wdefault)]
    The use of ‘forall’ as an identifier
    will become an error in a future GHC release.
    Suggested fix:
      Consider using another name, such as
      ‘forAll’, ‘for_all’, or ‘forall_’.
